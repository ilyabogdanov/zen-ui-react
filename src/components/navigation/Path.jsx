import React from "react";

/**
 * @class
 * @classdesc Path
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {*} props.children
 */
export default class Path extends React.Component {
    render() {
        const { children } = this.props;
        return (
            <div className="zen_ui__path">
                <div>
                    <div>
                        {children}
                    </div>
                </div>
            </div>
        );
    }
};