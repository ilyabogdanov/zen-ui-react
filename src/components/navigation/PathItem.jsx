import React from "react";
import { Link } from "react-router-dom";

/**
 * @class
 * @classdesc Path item
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {object|string} props.to
 * @prop {boolean} props.replace
 * @prop {function} props.innerRef
 *
 * @prop {*} props.children
 */
export default class PathItem extends React.Component {
    render() {
        const { to, replace, innerRef, children } = this.props;
        return (
            <span className="zen_ui__path_item">
                {to ? <Link to={to} replace={replace} innerRef={innerRef}>{children}</Link> : <span>{children}</span>}
            </span>
        );
    }
};