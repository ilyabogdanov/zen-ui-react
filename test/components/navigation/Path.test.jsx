import React from "react";
import { BrowserRouter, StaticRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Path from "../../../src/components/navigation/Path";
import PathItem from "../../../src/components/navigation/PathItem";

describe("Path", () => {
    test("should render correctly", () => {
        const e = renderer.create(
            <BrowserRouter>
            <StaticRouter>
            <Path>
                <PathItem to="/"/>
                <PathItem to="/"/>
                <PathItem to="/"/>
                <PathItem/>
            </Path>
            </StaticRouter>
            </BrowserRouter>
        ).toJSON();
        expect(e).toMatchSnapshot();
    });
});
