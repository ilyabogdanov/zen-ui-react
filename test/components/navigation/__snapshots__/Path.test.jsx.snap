// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path should render correctly 1`] = `
<div
  className="zen_ui__path"
>
  <div>
    <div>
      <span
        className="zen_ui__path_item"
      >
        <a
          href="/"
          onClick={[Function]}
        />
      </span>
      <span
        className="zen_ui__path_item"
      >
        <a
          href="/"
          onClick={[Function]}
        />
      </span>
      <span
        className="zen_ui__path_item"
      >
        <a
          href="/"
          onClick={[Function]}
        />
      </span>
      <span
        className="zen_ui__path_item"
      >
        <span />
      </span>
    </div>
  </div>
</div>
`;
